tracer/src/Datadog.Trace/Debugger/Helpers/MethodExtensions.cs

```diff
@@ -0,0 +1,183 @@
+// <copyright file="MethodExtensions.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Datadog.Trace.Debugger.Helpers
+{
+    internal static class MethodExtensions
+    {
+        /// <summary>
+        /// Gets fully qualified name of a method with parameters and generics. For example SkyApm.Sample.ConsoleApp.Program.Main(String[] args).
+        /// Code was copied from System.Diagnostics.StackTrace.ToString() - .NET Standard implementation, not .NET Framework
+        /// </summary>
+        internal static string GetFullyQualifiedName(this MethodBase mb)
+        {
+            try
+            {
+                var sb = new StringBuilder(255);
```

Comment:
255 seems potentially quite small, especially if you have generics? Would probably be better to use the `StringBuilderCache`?

Also, FYI, we potentially already have a bunch of this code in `ExceptionRedactor`, which similarly was based on `StackTrace.ToString()`!

Reply:
Thanks. I will look into `ExceptionRedactor` and see if there's something I can extract. For the time being, I've based the usage on `StringBuilderCache` in ed3fbbbe6771d64dd582dd2ed3be5de599705a0a.